pandas/core/dtypes/dtypes.py

```diff
@@ -13,10 +25,21 @@
 from pandas.core.dtypes.generic import ABCCategoricalIndex, ABCDateOffset, ABCIndexClass
 from pandas.core.dtypes.inference import is_bool, is_list_like
 
+if TYPE_CHECKING:
+    import pyarrow  # noqa: F401
+    from pandas.core.arrays import (  # noqa: F401
+        IntervalArray,
+        PeriodArray,
+        DatetimeArray,
+    )
+    from pandas import Categorical  # noqa: F401
+
 str_type = str
 
+ExtensionDtypeT = TypeVar("ExtensionDtypeT", bound=ExtensionDtype)
```

Comment:
When do you need to use `ExtensionDtypeT` and when can you use `ExtensionDtype` (both are used in this file/diff)

Reply:
not using a typevar for `find` may be an oversight, I'll look into whether necessary.

Reply:
the only callsites are in pandas\core\dtypes\common.py, so `find` may need typevars once types are added there, and the callsites to there are typed etc. etc. 

There does seem to be some resistance to typevars, shall I add now or wait till becomes necessity?

I've also just noticed that the type annotation for `dtype` in `find` is also incorrect...

the code is 
```
            if not isinstance(dtype, type):
                dtype_type = type(dtype)
            if issubclass(dtype_type, ExtensionDtype):
                return dtype
```

so dtype parameter can also accept an instance as well as a type. again could update here or in a subsequent PR. only the return type was changed in this PR as this caused errors once construct_from_string was typed as this returns an instance, return annotation on master is only 
 type.

Reply:
The typevar has been removed for now.